crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -476,13 +476,16 @@ fn normalize_comment<'a>(
 
         // Black adds a space before the non-breaking space if part of a type pragma.
         if trimmed.trim_start().starts_with("type:") {
-            return Ok(Cow::Owned(std::format!("# \u{A0}{trimmed}")));
+            return Ok(Cow::Owned(std::format!("# {content}")));
         }
 
         // Black replaces the non-breaking space with a space if followed by a space.
         if trimmed.starts_with(' ') {
             return Ok(Cow::Owned(std::format!("# {trimmed}")));
         }
+
+        // Otherwise we replace the first non-breaking space with a regular space.
+        return Ok(Cow::Owned(std::format!("# {}", &content["\u{A0}".len()..])));
```

Comment:
Nit: Github doesn't allows me to add a suggestion for the whole code block, but we can now rewrite this function to not use return

```
if content.starts_with('\u{A0}') {
	if trimmed.trim_start().starts_with("type") {
		...
	} else if trimmed.starts_with(' ') {
		...
	} else {
		...
	}
} else {
	... 
}
```

Reply:
Like this? 1e92b1c01ac8882db0d4d74865cd4c03eb9217df